Detectors/Base/include/DetectorsBase/Vector.h

```diff
@@ -0,0 +1,393 @@
+    exit(1);      
+  }
+
+  // create a copy of the buffer
+  mPtr = std::make_unique<char[]>(nbDecoded);
+  std::copy(ptr,ptr+nbDecoded,mPtr.get());
+  
+}
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::adopt(std::unique_ptr<char[]>& ptr, sizeType nbytes)
+{
+  /**
+   * recreates container buffer from provided pointer assuming its ownership
+   * ptr:    pointer extracted from the container of simular type using e.g. release() methods
+   * nbytes: expected size of the buffer in bytes, used for error check
+   */
+  
+  try {
+    if (!ptr) {
+      throw "invalid arguments:";
+    }
+  } catch(const char* msg) {
+    LOG(FATAL) << msg << " rawPtr is null" << FairLogger::endl;
+  }
+  try {
+    if (nbytes>=0 && (nbytes<dataOffset())) {
+      throw "invalid arguments:";
+    }
+  } catch(const char* msg) {
+    LOG(FATAL) << msg <<  "wrong size " << nbytes <<  " at least " << dataOffset() << " expected" << FairLogger::endl;
+  }
+  
+  sizeType nbDecoded = (reinterpret_cast<Header*>(ptr.get()))->sizeInBytes;
+  try {
+    if (nbytes>=0 && (nbytes!=nbDecoded)) {
+      throw "invalid arguments:";
+    }
+  } catch(const char* msg) {
+    LOG(FATAL) << msg <<" supplied size " << nbytes <<  " differs from decoded size " << nbDecoded << FairLogger::endl;
+    exit(1);      
+  }
+
+  mPtr = std::move(ptr); // transfer ownership
+  
+}
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::reserve(sizeType n)
+{
+  /**
+   * Resize container to size n. 
+   * Existing content is preserved, but truncated to n if n<current_n_objects
+   */
+  if (n<0) n = 0;
+  sizeType bookSize = dataOffset() + n*sizeof(T);
+  //  auto tmpPtr = std::move(mPtr);
+  std::unique_ptr<char[]> tmpPtr = std::make_unique<char[]>(bookSize);
+  if (mPtr!=nullptr) std::copy(mPtr.get(), mPtr.get()+std::min(bookSize,sizeInBytes()), tmpPtr.get());  
+  mPtr.swap(tmpPtr);
+  if (n<getHeader()->nObjects) getHeader()->nObjects = n;
+  getHeader()->sizeInBytes = bookSize;
+}
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::setUserInfo(const H& v)
+{
+  /**
+   * Sets data identification field
+   */
+  getHeader()->userInfo = v;
+  //  std::copy_n(&v,1, &getHeader()->userInfo);
+}
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::expand()
+{
+  /**
+   * expand container according to expansion policy
+   */
+  auto oldSize = capacity();
+  auto newSize = getExpandPolicy()<0 ?
+    2*std::max(oldSize-getExpandPolicy(),static_cast<sizeType>(1)) :
+    oldSize+getExpandPolicy();
+  reserve(newSize);
+}
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::clear(bool calldestructor)
```

Comment:
I think "calldestructor" is unnecessary and it reminds me too much of ROOT IsOwner()-like semantics - which go against a clearly defined ownership model. Like in std::vector you either always destroy if you contain objects directly, or never it you have pointers. I think here we by construction never have to/want to destroy:
by construction T is_trivially_copyable which (I think at least in our case) implies that the dtor has no side effects (since you can just memcpy it and T cannot have pointer members etc.) - so calling the dtor should never be needed.